packages/http/src/util.js

```diff
@@ -51,23 +51,37 @@ const assertUrl = (pathOrUrl, baseUrl) => {
   }
 };
 
+export const CONTENT_TYPES = {
+  xml: 'application/xml',
+  json: 'application/json',
+  string: 'text/plain',
+};
+
 /**
  * Request helper function
  * @function
  * @private
  */
-export function request(method, path, params, callback = s => s) {
+export function request(method, path, params) {
   return state => {
     const [resolvedPath, resolvedParams = {}] = expandReferences(
       state,
       path,
       params
     );
 
-    let { body, headers = { 'Content-Type': 'application/json' } } = resolvedParams;
+    let { body, contentType = 'json', headers = {} } = resolvedParams;
 
-    if (resolvedParams.form) {
+    const hasContentType = Object.keys(headers).some(
```

Comment:
Aha! This stuff shouldn't be here - we have a separate PR for this. It shouldn't matter but there's a chance of introducing a difficult conflict